draft-ietf-quic-tls.md

```diff
@@ -644,6 +647,27 @@ A client MAY attempt to send 0-RTT again if it receives a Retry or Version
 Negotiation packet.  These packets do not signify rejection of 0-RTT.
 
 
+## Validating 0-RTT Configuration
+
+When a server receives a ClientHello with the "early_data" extension, it has to
+decide whether to accept or reject early data from the client. Some of this
+decision is made by the TLS stack (e.g., checking that the cipher suite being
+resumed was included in the ClientHello; see Section 4.2.10 of {{!TLS13}}). Even
+when the TLS stack has no reason to reject early data, the QUIC stack or the
+application protocol using QUIC might reject early data because the
+configuration of the transport or application associated with the resumed
+session is not compatible with the server's current configuration.
+
+QUIC requires additional transport state to be associated with a 0-RTT session
+ticket. If stateless session tickets are used, this information must be stored
+in the session ticket. Application protocols that use QUIC might have similar
```

Comment:
While it is true that additional transport state needs to be associated and verified when doing 0-RTT resumption, I am not sure if that necessarily mean that "this information must be stored in the session ticket."

For example, a server might store the associated states in a NEW_TOKEN token, that also contains the hash value of the session ticket. In such a design, when accepting a new connection, a server can consult the pair of the token and the session ticket, and decide if it can accept 0-RTT.

Reply:
@nharper : Can you address this comment?

Reply:
Would it work to change "If stateless session tickets are used, this information must be stored in the session ticket." to "If stateless session tickets are used, this information may be stored in the session ticket."?

Reply:
That would be fine.  The use of "must", implying "MUST", is a little too strong.

I would probably say that the choice of location (NEW_TOKEN vs NST) is an implementation-dependent one.  Implementations might want to use NST as that has better fate-sharing and they might not want to concern themselves with path validation for 0-RTT (or to allow a limited amount of 0-RTT in the case the token is bad).  Or, they could insist on a token and strong path validation, so moving the larger state necessary for 0-RTT into the token makes more sense.

Reply:
The intention of that sentence was to say that most likely you're using stateless tickets and storing this additional state in the ticket, so I reworded it to say that.